File: backend/app/schemas/mentor.py
"""师徒 HTTP DTO（M7 L6）。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class MentorApplyRequest(BaseModel):
    """拜师 / 收徒申请。"""

    target_character_id: int | None = None
    target_name: str | None = None
    intent: str = Field(description="apprentice=拜师；master=收徒")


class MentorLessonRequest(BaseModel):
    """日课三选一：传道 / 授业 / 解惑。"""

    kind: str = Field(description="dao|craft|technique")
    resource: str | None = Field(
        default=None,
        description="传道时 spirit|body",
    )
    target_id: str | None = Field(
        default=None,
        description="授业/解惑目标功法 id",
    )


class MentorTeachRequest(BaseModel):
    """传授功法或配方图纸。"""

    item_kind: str = Field(description="technique|recipe")
    item_id: str = Field(description="功法或配方 id")


class MentorStudyRequest(BaseModel):
    """徒弟请学师傅功法。"""

    technique_id: str = Field(description="师傅已掌握的功法 id")


class MentorDirectRequest(BaseModel):
    """师傅设置亲传弟子。"""

    apprentice_character_ids: list[int] = Field(
        default_factory=list,
        description="亲传弟子角色 id（最多配置名额，可含已出师）",
    )


class FerryRescueRequest(BaseModel):
    """社交引渡。"""

    mode: str = Field(description="friend|sect|kin（普渡/道友、同门、亲友）")
    target_character_id: int | None = None
    target_name: str | None = None

File: frontend/src/api/mentor.ts
/**
 * M7 L6 师徒 HTTP API。
 */
import { http } from './http'
import { envelopeFromAxiosError } from './envelope'
import type { ApiResponse } from '../types/api'
import type { MentorMePayload } from '../types/mentor'

export async function fetchMentorMe(): Promise<ApiResponse<MentorMePayload>> {
  try {
    const response = await http.get<ApiResponse<MentorMePayload>>('/mentor/me')
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError<MentorMePayload>(error)
  }
}

export async function applyMentor(body: {
  target_name?: string | null
  target_character_id?: number | null
  intent: 'apprentice' | 'master'
}): Promise<ApiResponse<{ message?: string; bond_id?: number }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string; bond_id?: number }>>(
      '/mentor/apply',
      body,
    )
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function acceptMentor(bondId: number): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>(
      `/mentor/${bondId}/accept`,
    )
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function rejectMentor(bondId: number): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>(
      `/mentor/${bondId}/reject`,
    )
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function teachLesson(body: {
  kind: 'dao' | 'craft' | 'technique'
  resource?: 'spirit' | 'body' | null
  target_id?: string | null
}): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/lesson', body)
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function teachItem(body: {
  item_kind: 'technique' | 'recipe'
  item_id: string
}): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/teach', body)
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function studyTechnique(body: {
  technique_id: string
}): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/study', body)
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function setDirectDisciples(body: {
  apprentice_character_ids: number[]
}): Promise<
  ApiResponse<{ message?: string; log_lines?: string[]; appointed?: string[]; cleared?: string[] }>
> {
  try {
    const response = await http.post<
      ApiResponse<{
        message?: string
        log_lines?: string[]
        appointed?: string[]
        cleared?: string[]
      }>
    >('/mentor/direct', body)
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

/** @deprecated 兼容旧传功；等价传道·修为 */
export async function passCultivation(): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/pass')
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function graduateMentor(): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/graduate')
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}

export async function dissolveMentor(): Promise<ApiResponse<{ message?: string }>> {
  try {
    const response = await http.post<ApiResponse<{ message?: string }>>('/mentor/dissolve')
    return response.data
  } catch (error: unknown) {
    return envelopeFromAxiosError(error)
  }
}
